### ISSUE TITLE
"Types" not colored properly

### ISSUE THREAD

**Issue by felixfbecker** [Microsoft/TypeScript-TmLanguage#114]
![image](https://cloud.githubusercontent.com/assets/10532611/13599312/78a6c28e-e522-11e5-9017-2d53f7a24aaa.png)

```ts
export type BreakpointType = 'line' | 'call' | 'return' | 'exception' | 'conditional' | 'watch'
```

directly using the string literal types works, the issue appears only when using types.

VS Code February.

**Comment by Ciantic** [comment]
I wonder how these should be colored? I've been looking at the TmLanguage syntax, and since it's ungodly mess of regexps it can't for instance color interfaces, or properties with one color consistently.

I don't think one can color string literals with specific color using TmLanguage

**Comment by felixfbecker** [comment]
@Ciantic the following code snippet is working fine:
```ts
interface Test {
  type: 'line' | 'call' | 'return';
}
```
the issue only happens when using `type` keyword to alias a custom type.

**Comment by Ciantic** [comment]
I would argue that since TmLanguage is all regexps, it shouldn't try to be this smart. Right now with latest styles I get this:

![image](https://cloud.githubusercontent.com/assets/64731/13888473/517c1652-ed4a-11e5-8d39-18101acd663e.png)

Which is really broken.

Clearly there is too much regexp smarts here, if more smarts was required this should be implemented with scanner in TS compiler: https://basarat.gitbooks.io/typescript/content/docs/compiler/scanner.html

It even gives the location of things in a file such an easily.

**Comment by DanielRosenwasser** [comment]
It should just colorize those as string literals. Not as specialized types.

**Comment by felixfbecker** [comment]
@Ciantic there is not a problem with regexp smarts here, this is just how textmate grammars work. You could not parse Typescript with just a bunch of regexps. Textmate language files are a way to describe (context-free) grammars. Regexps are only used to describe the production rules.

**Comment by Ciantic** [comment]
Btw, mr Basarat wrote up on implementation of Syntax Highlighting based on compiler in here:

https://github.com/alm-tools/alm/blob/master/docs/contributing/SYNTAX.md

https://github.com/alm-tools/alm/tree/master/docs/features#syntax-highlighting

I think Microsoft should just hire the man, he seems to be pulling all the stops for improving TS development.

**Comment by sheetalkamat** [comment]
Fixed by #257
